File: src/main/java/me/chriss99/glabstractions/DoubleGLBuffer.java
package me.chriss99.glabstractions;

import static org.lwjgl.opengl.GL11.GL_DOUBLE;
import static org.lwjgl.opengl.GL15.glBufferData;

public class DoubleGLBuffer extends GLBuffer {
    public DoubleGLBuffer(int target) {
        super(target, GL_DOUBLE);
    }

    public DoubleGLBuffer updateData(double[] data, int usage) {
        bind();
        glBufferData(target, data, usage);
        return this;
    }
}

File: src/main/kotlin/glabstractions/DoubleGLBuffer.kt
package me.chriss99.glabstractions

import org.lwjgl.opengl.GL11.GL_DOUBLE
import org.lwjgl.opengl.GL15.glBufferData

class DoubleGLBuffer(target: Int) : GLBuffer(target, GL_DOUBLE) {
    fun updateData(data: DoubleArray, usage: Int): DoubleGLBuffer {
        bind()
        glBufferData(target, data, usage)
        return this
    }
}